Client/game_sa/CWaterManagerSA.cpp

```diff
@@ -742,17 +742,25 @@ bool CWaterManagerSA::SetPositionWaterLevel(const CVector& vecPosition, float fL
     return SetPolyWaterLevel(pPoly, fLevel, pChangeSource);
 }
 
-bool CWaterManagerSA::SetWorldWaterLevel(float fLevel, void* pChangeSource, bool bIncludeWorldNonSeaLevel)
+bool CWaterManagerSA::SetWorldWaterLevel(float fLevel, void* pChangeSource, bool bIncludeWorldNonSeaLevel, bool bIncludeWorldSeaLevel, bool bIncludeOutsideWorldLevel)
 {
     assert(m_bInitializedVertices);
     CVector vecVertexPos;
-    for (DWORD i = 0; i < NUM_DefWaterVertices; i++)
+
+    if (bIncludeWorldSeaLevel || bIncludeWorldNonSeaLevel)
     {
-        m_Vertices[i].GetPosition(vecVertexPos);
-        if (bIncludeWorldNonSeaLevel || !m_Vertices[i].IsWorldNonSeaLevel())
-            vecVertexPos.fZ = fLevel;
-        m_Vertices[i].SetPosition(vecVertexPos, pChangeSource);
+        for (DWORD i = 0; i < NUM_DefWaterVertices; i++)
+        {
+            m_Vertices[i].GetPosition(vecVertexPos);
+            if ((bIncludeWorldNonSeaLevel && m_Vertices[i].IsWorldNonSeaLevel()) || (bIncludeWorldSeaLevel && !m_Vertices[i].IsWorldNonSeaLevel())) 
+                vecVertexPos.fZ = fLevel;
+            m_Vertices[i].SetPosition(vecVertexPos, pChangeSource);
+        }        
```

Comment:
Not sure what's the type of `m_Vertices`, but Im pretty sure you could use a for based range loop (eg.: `for(auto& vertex : m_Vertices)`

Reply:
NUM_DefWaterVertices is constant value, I think that i will leave this.